libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -710,7 +723,8 @@ impl NodeAPI for Greenlight {
 
         // calculate onchain balance
         let onchain_balance = self.on_chain_balance(funds.clone()).await?;
-        let utxos = self.utxos(funds).await?;
+        let pending_onchain_balance = self.pending_onchain_balance(funds.clone()).await?;
```

Comment:
I think we should be able to borrow `funds`, so it doesn't need to be cloned.
```suggestion
        let pending_onchain_balance = self.pending_onchain_balance(&funds).await?;
```

Reply:
Done, should I add this for the `onchain_balance` as well?

Reply:
Added it for `onchain_balance` too lmk if I should drop this commit.

Reply:
Adding to onchain_balance too is great 👍